ooniprobe/View/TestResults/Footer/UploadFooterViewController.m

```diff
@@ -107,7 +108,11 @@ -(void)uploadMeasurements:(NSArray *)notUploaded startAt:(NSUInteger)idx{
 
 -(BOOL)uploadMeasurement:(Measurement*)measurement{
     NSString *content = [TestUtility getUTF8FileContent:[measurement getReportFile]];
+    NSUInteger bytes = [content lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
     MKCollectorResubmitSettings *settings = [[MKCollectorResubmitSettings alloc] init];
+    //Timeout dependent on the body size considering a minimum upload speed of 12.5 kbit/s + 10 s
+    NSUInteger timeout = bytes * 8 / (12.5 * 1000) + 10;
+    [settings setTimeout:timeout < STANDARD_TIMEOUT ? STANDARD_TIMEOUT : timeout];
```

Comment:
Strictly speaking we should check that bytes * 8 does not overflow. It would be better to avoid multiplying by `8` and represent the speed in `kbit/s` so we cannot overflow.

Reply:
I would like this functionality to be a static method and I would like to see tests for some corner cases plus a common case.